src/odh-uploader/handler.test.ts

```diff
@@ -1,27 +1,22 @@
-import AWS from 'aws-sdk';
 import { S3Event } from 'aws-lambda';
 import nodemailer from 'nodemailer';
 import * as testData from './testData/testData';
 import { createMailTransporter, odhUploaderHandler } from './handler';
+import * as s3 from './data/s3';
 
 jest.mock('aws-sdk');
 
 describe('odhHandler SES emailer', () => {
-    const mockS3GetObject = jest.fn();
     const mockMailTransporter = jest.fn();
 
-    beforeEach(() => {
-        (AWS.S3 as {}) = jest.fn().mockImplementation(() => {
-            return {
-                getObject: mockS3GetObject,
-            };
-        });
+    const mockFetchDataFromS3Spy = jest.spyOn(s3, 'fetchDataFromMatchingS3');
+    const mockGetNetexFileFromS3 = jest.spyOn(s3, 'getNetexFileFromS3');
 
-        mockS3GetObject.mockImplementation(() => ({
-            promise(): Promise<{}> {
-                return Promise.resolve({ Body: testData.testNetexFromS3 });
-            },
-        }));
+    beforeEach(() => {
+        mockFetchDataFromS3Spy.mockImplementation(() =>
+            Promise.resolve({ Body: testData.testNetexFromS3, email: 'test@example.com' }),
+        );
+        mockGetNetexFileFromS3.mockImplementation(() => Promise.resolve('Body: testData.testNetexFromS3'));
```

Comment:
I think we need to update these mock Promise.resolve() objects. The mockFetchDataFromS3Spy will return a matching data object as far as I can tell? And the mockGetNetexFileFromS3Spy will return a just a string of netex?

Reply:
sorted using the matching test data